internal/lib/utils.go

```diff
@@ -546,3 +547,31 @@ func IsInfraSettingNSScoped(infraSetting, namespace string) bool {
 	}
 	return false
 }
+
+type LockSet struct {
+	locks map[string]*sync.Mutex
```

Comment:
we can use sync.map here and do away with explicit lock/unlock logic.

Reply:
all threads are required to perform atomic updates. sync.map will not work for this use case

Reply:
ok, in that case let's use sync.map instead of regular map which would not be safe for concurrent write/read

Reply:
replaced regular golang map with sync.Map, thanks